### PR TITLE
usage: Implementation for usage plugin api change. See #1468

### PR DESCRIPTION
**Pull request by sbrossie**
This depends on https://github.com/killbill/killbill-plugin-api/pull/21

Also, we should decide if we also want to change our [usage endpoint](https://github.com/killbill/killbill/blob/killbill-0.22.28/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/UsageResource.java#L173) along with [usage api](https://github.com/killbill/killbill-api/blob/master/src/main/java/org/killbill/billing/usage/api/UsageUserApi.java#L52) to pass an additional `pluginProperty` and pass it all the way to the usage plugin.

### CODE REVIEW

File: usage/src/main/java/org/killbill/billing/usage/api/BaseUserApi.java [hunk 1]
```diff
@@ -51,16 +52,16 @@ public BaseUserApi(final OSGIServiceRegistration<UsagePluginApi> pluginRegistry)
...
-    protected List<RawUsageRecord> getSubscriptionUsageFromPlugin(final UUID subscriptionId, final LocalDate startDate, final LocalDate endDate, final TenantContext tenantContext) {
-        return getUsageFromPlugin(subscriptionId, startDate, endDate, tenantContext);
+    protected List<RawUsageRecord> getSubscriptionUsageFromPlugin(final UUID subscriptionId, final LocalDate startDate, final LocalDate endDate, final UsageContext usageContext) {
+        return getUsageFromPlugin(subscriptionId, startDate, endDate, usageContext);
     }
 
-    private List<RawUsageRecord> getUsageFromPlugin(@Nullable final UUID subscriptionId, final LocalDate startDate, final LocalDate endDate, final TenantContext tenantContext) {
-        Preconditions.checkNotNull(tenantContext.getAccountId(), "TenantContext has no accountId");
+    private List<RawUsageRecord> getUsageFromPlugin(@Nullable final UUID subscriptionId, final LocalDate startDate, final LocalDate endDate, final UsageContext usageContext) {
+        Preconditions.checkNotNull(usageContext.getAccountId(), "TenantContext has no accountId");
```
Comment (pierre) [inline usage/src/main/java/org/killbill/billing/usage/api/BaseUserApi.java:64]: Nit: fix comment (`UsageContext`).

File: usage/src/main/java/org/killbill/billing/usage/api/DefaultUsageContext.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Override
+    public UUID getAccountId() {
+        return context.getAccountId();
+    }
+
+    @Override
+    public UUID getTenantId() {
+        return context.getTenantId();
+    }
```
Comment (pierre) [inline usage/src/main/java/org/killbill/billing/usage/api/DefaultUsageContext.java:57]: Nit: add `toString(), equals(), hashCode()`

**Review by pierre — APPROVED** [review]
As discussed, 👍 to update the GET path.

**Comment by sbrossie** [comment]
> As discussed, 👍 to update the GET path.

Fixed in https://github.com/killbill/killbill/pull/1648/commits/18775bfb4b14e0748f3391d3fc9379ab28e9cf92 and See https://github.com/killbill/killbill-api/pull/88

File: usage/src/main/java/org/killbill/billing/usage/api/BaseUserApi.java (hunk 1 above)
Comment (sbrossie) [reply usage/src/main/java/org/killbill/billing/usage/api/BaseUserApi.java]: Fixed in https://github.com/killbill/killbill/pull/1648/commits/eaae451610ed4d3fbe6cafc9dfdf65e76de4cd5d

File: usage/src/main/java/org/killbill/billing/usage/api/DefaultUsageContext.java (hunk 2 above)
Comment (sbrossie) [reply usage/src/main/java/org/killbill/billing/usage/api/DefaultUsageContext.java:58]: Fixed in https://github.com/killbill/killbill/pull/1648/commits/eaae451610ed4d3fbe6cafc9dfdf65e76de4cd5d

**Review by pierre — APPROVED** [review]

**Pull request merged**